### PR TITLE
[client][c++] add getLastMessageIdAsync in Consumer

### PR DESCRIPTION
**Pull request by komalatammal**
### Motivation

Add getLastMessageId method to C++ Consumer.cc to address the missing part in this PR https://github.com/apache/pulsar/pull/15993

### Modifications

Expose methods to get last message Id in consumer, the C++ client's Consumer class

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (yes)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)
  - 
### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why) 
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: pulsar-client-cpp/include/pulsar/Consumer.h
Comment (BewareMyPower) [inline pulsar-client-cpp/include/pulsar/Consumer.h:397]:
```suggestion
    void getLastMessageIdAsync(GetLastMessageIdCallback callback);
```

This method definition is already in namespace pulsar.

**Review by BewareMyPower — APPROVED** [review]

**Comment by eolivelli** [comment]
@BewareMyPower we need 2 bindings +1 in order to commit a patch.
we should have waited for a second reviewer

**Comment by BewareMyPower** [comment]
Okay. Got it.

**Comment by BewareMyPower** [comment]
@eolivelli But I have a concern that should it be a strict constraint or a blocker? Or how long should we wait if no one has time to review or doesn't want to review? I'm afraid currently there are not enough reviewers for all PRs. For example, https://github.com/apache/pulsar/pull/16072 is another PR I've merged with only 1 binding +1. But no one reviewed this PR in 5 days after it's opened, though I've pinged reviewers again.

**Comment by eolivelli** [comment]
yes, it is a strict constraint for all non-trivial changes.

if you want to sponsor a patch the best way is to send a message to dev@ and ask for a second reviewer.

5 days it not much time in a OSS community, we have to keep in mind that committers are volunteers.
there is no hurry in committing a patch usually and we can always ping the community if we think that a patch is really important or urgent.

**Comment by BewareMyPower** [comment]
Okay, I'll keep it in mind next time.

**Comment by michaeljmarshall** [comment]
> yes, it is a strict constraint for all non-trivial changes.

@eolivelli - I did not realize this was a strict requirement. Have we discussed this on the dev mailing list recently? It might be worth re-iterating the requirement. It might also be worth documenting here: https://github.com/apache/pulsar/wiki/Committer-Guide.

**Pull request merged**
